### ISSUE TITLE
Segmentation fault when running an app that uses a C++ module

### ISSUE THREAD

**Issue by miki-c** [vercel/ncc#709]
### Normal behaviour (without ncc)
I am writing a Node application that imports a function from a C++ module (also written by me), by using `n-api `(https://nodejs.org/api/n-api.html). 

The C++ module is compiled by running `node-gyp configure && node-gyp build`. Upon successful compilation, several files are generated and placed in the `build/` folder. Most importantly, I can import the C++ module in Node like so:
`const addon = require('./my-addon/build/Release/addon');`
and then invoke the exposed function like so:
`addon.myCppFunction("hello");`

Everything works flawlessly so far, as I am able to run the application which imports and invokes the C++ function as expected.

### The problem (with ncc)
At this point, the C++ module has already been compiled. So I can compile the whole Node application with `ncc`, which generates two output files in the target directory: `index.js` and `addon.node`.
Though, running `index.js` causes a _Segmentation fault_ as soon as the C++ function is invoked. The logged output is not very helpful, all it says is literally "_Segmentation fault_".  I placed a `printf` at the beginning of the C++ function, and that gets printed. So at least I know that the function is correctly imported and invoked.

Is this a bug? Or am I missing a step when using `ncc` to include the C++ module?

### Specs 
- `ncc` version: `"@vercel/ncc": "^0.25.1"`
- OS version (actually a Docker image): `node:14.15.0-alpine3.12`

**Comment by styfle** [comment]
A couple things to try:

- ncc may have missed some files besides `index.js` and `addon.node` so verify if that is the case
- try running your original app on alpine without ncc (it sounds like you tried but just want to double check since that would rule out ncc)

**Comment by miki-c** [comment]
Here is a further test I just carried out:

1. In the first stage of the Dockerfile, I compiled the C++ module with `node-gyp`
2. In the second stage of the Dockerfile, I copied these two files in a folder: 
- `my-addon/build/Release/addon.node` (ie, the compiled C++ module) 
- `test-addon.js` (ie, the node program which simply imports and invokes the C++ function) 
3. I started the container, opened a shell in it, navigated to the folder of interest and ran: `node test-addon.js` which worked as expected. 

- Bear in mind that I only copied the two files above in this folder. There is nothing else in it, not even `node_modules`. This demonstrates that those two files are sufficient for the program to run, and no other files are needed.

4. From the shell, in the same folder, I installed ncc by running: `npm i @vercel/ncc@0.25.1`
5. Then, I bundled the app with ncc by running: `npx ncc build test-addon.js -o dist/`
6. Lastly, I ran `node dist/index.js` and, as soon as the C++ function was invoked, a _Segmentation fault_ error was thrown.

**Comment by miki-c** [comment]
I rewrote the whole thing from scratch. After running through every single step, I finally found a bug in our C++ module (basically, my coworker forgot to initialise a variable). So it was not a bug with `ncc` after all, sorry guys.
Now, an interesting question would be: "Why did the segmentation fault only happen when the app was bundled with `ncc`?". Well, I have no idea but that's a question for another time. I think I've spent enough hours on this already :D
